src/main/kotlin/org/radarbase/push/integration/garmin/service/BackfillService.kt

```diff
@@ -107,10 +107,23 @@ class BackfillService(
         logger.debug("Making Request: {}", req.request)
         try {
             httpClient.newCall(req.request).execute().use { response ->
-                if (response.isSuccessful) {
-                    requestGenerator.requestSuccessful(req, response)
-                } else {
-                    requestGenerator.requestFailed(req, response)
+                when {
+                    response.isSuccessful -> requestGenerator.requestSuccessful(req, response)
+                    response.code == 429 -> {
+                        requestGenerator.requestFailed(req, response)
+                        logger.info(
+                            "Too many requests for user ${req.user.id}." +
+                                " Backing off for ${BACKOFF_TIME_MS / 1000}s...."
+                        )
+                        Thread.sleep(BACKOFF_TIME_MS)
```

Comment:
It might make sense to introduce this timeout inside the request generator. This way, the current thread will be blocked for all users.